llumnix/entrypoints/vllm/api_server.py

```diff
@@ -224,35 +224,44 @@ async def is_ready():
     ready_status = await engine_manager.is_ready.remote()
     return ready_status
 
+class CustomArgumentParser(argparse.ArgumentParser):
+    def __init__(self, *args, **kwargs):
+        self.cur_namespace = "llumnix"
+        super().__init__(*args, **kwargs)
+
+    def set_namespace(self, namespace: str):
+        self.cur_namespace = namespace
+
+    def add_argument(self, *args, **kwargs):
+        if self.cur_namespace == 'llumnix' and "--help" not in args and args[0] != "--config-file":
```

Comment:
why args[0] != "--config-file" rather than args[0] == "--config-file"?

Reply:
args[0] != "--config-file" ensures that the code allows a specific exception for the --config-file option while enforcing a rule (default must be None) for other arguments